src/cryptonote_core/service_node_rules.h

```diff
@@ -113,6 +113,20 @@ namespace service_nodes {
   constexpr std::array<int, 3> MIN_STORAGE_SERVER_VERSION{{2, 0, 0}};
   constexpr std::array<int, 3> MIN_LOKINET_VERSION{{0, 6, 1}};
 
+  // The minimum accepted version number, broadcasted by Service Nodes via uptime proofs for each hardfork
+  struct proof_version
+  {
+    uint8_t hardfork;
+    std::array<uint16_t, 3> version;
+  };
+
+  constexpr proof_version MIN_UPTIME_PROOF_VERSIONS[] = {
+    {cryptonote::network_version_15_lns,                  {7,1,0}},
```

Comment:
This will break testnet as-is; we should leave it at 7,0,0 for now and bump it to 7,1,0 with the mainnet-ready version bump.

Reply:
Hm I wasn't gonna merge until 7.1.0

Reply:
Actually we might need for testnet statistics.